File: 21551-naveen-siva-kumar/week7/141-linked-list-cycle/141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode* fastPointer = head;
        ListNode* slowPointer = head;
        
        if(head==NULL || head->next == NULL){ return false;}
        if(head->next->next == NULL){return false;}
        
        while(fastPointer!=NULL || fastPointer->next!=NULL ){
            
            fastPointer =  fastPointer->next->next;
            slowPointer = slowPointer->next; 
            
            if(fastPointer==NULL || fastPointer->next==NULL ||  fastPointer->next->next == NULL){
                return false;
            }
            if(fastPointer == slowPointer) { 
                cout<<fastPointer;cout<<slowPointer;
                return true;}
            
                
        }
        return false;
    }
};

File: 21557 - Mukesh J/Week-7/141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode slow=head, fast=head;
        if(head==null)
            return false;
        while(fast!=null)
        {
            if(fast.next!=null && fast.next.next!=null)
            {
                fast= fast.next.next;
                slow= slow.next;
                if(fast==slow)
                    break;
            }
            else
                return false;
            
        }
        return true;
    }
}